src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/RecordComponentNumberCheck.java

```diff
@@ -0,0 +1,356 @@
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_PROTECTED_COMPONENTS = "too.many.protectedComponents";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_PUBLIC_COMPONENTS = "too.many.publicComponents";
+
+    /** Default maximum number of allowed components. */
+    private static final int DEFAULT_MAX_COMPONENTS = 8;
+
+    /** Specify the maximum number of {@code private} components allowed. */
+    private int maxPrivate = DEFAULT_MAX_COMPONENTS;
+    /** Specify the maximum number of {@code package} components allowed. */
+    private int maxPackage = DEFAULT_MAX_COMPONENTS;
+    /** Specify the maximum number of {@code protected} components allowed. */
+    private int maxProtected = DEFAULT_MAX_COMPONENTS;
+    /** Specify the maximum number of {@code public} components allowed. */
+    private int maxPublic = DEFAULT_MAX_COMPONENTS;
+
+    /**
+     * Setter to specify the maximum number of {@code private} components allowed.
+     *
+     * @param value the maximum allowed.
+     */
+    public void setMaxPrivate(int value) {
+        maxPrivate = value;
+    }
+
+    /**
+     * Setter to specify the maximum number of {@code package} components allowed.
+     *
+     * @param value the maximum allowed.
+     */
+    public void setMaxPackage(int value) {
+        maxPackage = value;
+    }
+
+    /**
+     * Setter to specify the maximum number of {@code protected} components allowed.
+     *
+     * @param value the maximum allowed.
+     */
+    public void setMaxProtected(int value) {
+        maxProtected = value;
+    }
+
+    /**
+     * Setter to specify the maximum number of {@code public} components allowed.
+     *
+     * @param value the maximum allowed.
+     */
+    public void setMaxPublic(int value) {
+        maxPublic = value;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.RECORD_DEF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.RECORD_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        // Instantiate new component counter
+        final ComponentCounter counter = new ComponentCounter();
+
+        // Count record components in this record definition
+        countComponents(counter, ast);
+
+        // Check counter and report violations
+        checkCounter(counter, ast);
+    }
+
+    /**
+     * Determine the visibility modifier, then iterate through the record component list
+     * and increment the counter.
+     *
+     * @param counter the ComponentCounter instance for this check
+     * @param ast the ast to check
+     *
```

Comment:
Extra blank line in the end of the comment

Reply:
Fixed.